firebase/app/routes/auth/google.tsx

```diff
@@ -0,0 +1,33 @@
+import type { LoaderArgs } from "@remix-run/node";
+import { redirect } from "@remix-run/node";
+
+import { signInWithIdp } from "~/server/auth.server";
+import { commitSession, getSession } from "~/sessions";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  // https://developers.google.com/identity/protocols/oauth2/openid-connect
+  const url = new URL(request.url);
+  const code = url.searchParams.get("code");
+  const response = await fetch("https://oauth2.googleapis.com/token", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      client_id: process.env.GOOGLE_CLIENT_ID,
+      client_secret: process.env.GOOGLE_CLIENT_SECRET,
+      code,
+      redirect_uri: "http://localhost:3000/auth/google",
```

Comment:
I don't remember why this was hard coded to localhost. Perhaps something is needed here. Will need to test a deployed version.

Reply:
confirmed this is an issue - using firebase functions locally (:5002) will redirect to :3000